Test/EncapsulationTestAwareTrait.php

```diff
@@ -0,0 +1,277 @@
+<?php
+
+/*
+ * This file is part of the ONGR package.
+ *
+ * (c) NFQ Technologies UAB <info@nfq.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ONGR\ElasticsearchBundle\Test;
+
+/**
+ * A helper for testing setters/getters.
+ *
+ * Trait EncapsulationTestAwareTrait
+ */
+trait EncapsulationTestAwareTrait
+{
+    /**
+     * @var string[] List of fields that should not be checked for tests.
+     */
+    private $ignoredFields = [];
+
+    /**
+     * @var mixed Stub object to test.
+     */
+    private $stub = null;
+
+    /**
+     * @return mixed
+     */
+    public function getStub()
+    {
+        return $this->stub;
+    }
+
+    /**
+     * @param mixed $stub
+     */
+    public function setStub($stub)
+    {
+        $this->stub = $stub;
+    }
+
+    /**
+     * Returns list of fields to test. Works as data provider.
+     *
+     * @return array
+     */
+    abstract public function getFieldsData();
+
+    /**
+     * Returns entity class name.
+     *
+     * @return string
+     */
+    abstract public function getClassName();
+
+    /**
+     * Returns list of fields that should not be checked for tests.
+     *
+     * @return array
+     */
+    protected function getIgnoredFields()
+    {
+        return $this->ignoredFields;
+    }
+
+    /**
+     * Set list of fields that should not be checked for tests.
+     *
+     * @param string[] $fields
+     */
+    protected function setIgnoredFields(array $fields)
+    {
+        $this->ignoredFields = $fields;
+    }
+
+    /**
+     * Set list of fields that should not be checked for tests.
+     *
+     * @param string $field
+     */
+    protected function addIgnoredField($field)
+    {
+        $this->ignoredFields[] = $field;
+    }
+
+    /**
+     * Tests field setter and getter.
+     *
+     * @param string        $field
+     * @param null|string   $type
+     * @param null|string   $addMethod
+     * @param null|string   $removeMethod
+     * @param null|string[] $additionalSetter
+     *
+     * @dataProvider getFieldsData()
+     */
+    public function testSetterGetter(
+        $field,
+        $type = null,
+        $addMethod = null,
+        $removeMethod = null,
+        $additionalSetter = null
+    ) {
+        /** @var \PHPUnit_Framework_TestCase|EncapsulationTestAwareTrait $this */
+
+        $objectClass = $this->getClassName();
+
+        $setter = 'set' . ucfirst($field);
+        $getter = 'get' . ucfirst($field);
+
+        if ($type === 'boolean') {
+            $getter = 'is' . ucfirst($field);
+        }
+
+        $stub = $this->getStub();
+
+        if (!isset($stub)) {
```

Comment:
isset() will return FALSE if testing a variable that has been set to NULL (php.net manual).
$this->stub is set to null by default. In case we don't change $this->stub value !isset($this->stub) should generate true. Or am I getting something wrong ?